jupyterhub/handlers/base.py

```diff
@@ -977,13 +977,16 @@ def abort():
             # waiting_for_response indicates server process has started,
             # but is yet to become responsive.
             if spawner._spawn_pending and not spawner._waiting_for_response:
-                # still in Spawner.start, which is taking a long time
-                # we shouldn't poll while spawn is incomplete.
-                self.log.warning(
-                    "User %s is slow to start (timeout=%s)",
-                    user_server_name,
-                    self.slow_spawn_timeout,
-                )
+                # If slow_spawn_timeout is intentionally disabled then we
+                # don't need to log a warning, just return.
+                if self.slow_spawn_timeout > 0:
```

Comment:
Should we avoid even calling `with_timeout` if `slow_spawn_timeout` is <= 0? What do we do instead? Pass `finish_spawn_future` to `gen.maybe_future`? I'm new to the tornado library so I'm not sure what is best here.

Reply:
I think it might be simpler to keep the control flow this way (too many branches and conditions can make it harder to ensure everything gets cleaned up in every case), but if you want to add the short-circuit before the wait, I think it would look like:

```python
if self.slow_spawn_timeout == 0 and not finish_spawn_future.done():
        # nothing to wait for if slow_spawn_timeout is disabled
        return

try:
    await ...
```